File: suites/osprey-compute-kernels/reference/cases/fib/fib.c
/* Naive recursive Fibonacci — exercises raw function-call + recursion overhead. */
#include <stdio.h>
#include <stdint.h>

static int64_t add(int64_t a, int64_t b) { return a + b; }
static int64_t sub(int64_t a, int64_t b) { return a - b; }

static int64_t fib(int64_t n) {
    switch (n) {
        case 0: return 0;
        case 1: return 1;
        default: return add(fib(sub(n, 1)), fib(sub(n, 2)));
    }
}

int main(void) {
    printf("%lld\n", (long long)fib(35));
    return 0;
}

File: suites/osprey-compute-kernels/reference/cases/fib/fib.rs
// Naive recursive Fibonacci — exercises raw function-call + recursion overhead.
fn add(a: i64, b: i64) -> i64 {
    a + b
}

fn sub(a: i64, b: i64) -> i64 {
    a - b
}

fn fib(n: i64) -> i64 {
    match n {
        0 => 0,
        1 => 1,
        _ => add(fib(sub(n, 1)), fib(sub(n, 2))),
    }
}

fn main() {
    println!("{}", fib(35));
}
